### DISCUSSION TITLE
Is there a way to assign a minimum window size

### DISCUSSION THREAD

**Discussion by pranavchavda** [leftwm/leftwm#753] [Ideas]
Since moving to LeftWm a few weeks ago, I have been using it exclusively. It's a joy to use. I was wondering if a minimum window width can be assigned to the main window in the center main or the center main balanced layouts, if not, it would be a great feature to consider for a future release.

**Comment by hertg** [comment]
I'm not sure how you mean, could you illustrate? The size of the Center Window is defined by the `MainWindowWidth` which you can change via the commands `IncreaseMainWidth` and `DecreaseMainWidth`. It is a size in percentage, which defaults to 50% I think.

**Reply by pranavchavda** [reply]
I didn't know about MainWindowWidth, thanks for that! I will try it out.

**Reply by pranavchavda** [reply]
I misunderstood your reply earlier. My objective was to set the MainWindowWidth to 70% at the time of logging in to the desktop. 

I tried to set MainWindowWidth via leftwm-command or via keybind, and it didn't work, I see why on rereading your post. What I did instead, and it worked, is to use xdotool on the up file in my theme to press the keybinds for IncreaseMainWidth, so it sets the window to my desired size upon starting up.

It's a kind of workaround and requires an extra tool, but it works for me!

**Reply by hertg** [reply]
I see, just a quick question. I'm working on a new layout-system for leftwm, so i'm interested in use-cases like that just for my understanding. Do you want that specific 70% `MainWindowWidth` on all layouts, or do you prefer different values for different layouts? How would you feel about the possibility to "define your own layouts" in the config file, with custom default values for dynamic settings like `MainWindowWidth`? Or do you have any other inputs you want me to know?

The new layout-system most probably won't include solutions for all use-cases, but knowing the different expectations of users helps me to identify potential problems in the customizability of the new system.
